src/Lecture1.hs

```diff
@@ -1,122 +1,136 @@
-
-To complete exercises, you need to complete implementation and add
-missing top-level type signatures. You can implement any additional
-helper functions. But you can't change the names of the given
-functions.
-
-Comments before each function contain explanations and example of
-arguments and expected returned values.
-
-It's absolutely okay if you feel that your implementations are not
-perfect. You can return to these exercises after future lectures and
-improve your solutions if you see any possible improvements.
--}
-
+-- |
+-- Module                  : Lecture1
+-- Copyright               : (c) 2021-2022 Haskell Beginners 2022 Course
+-- SPDX-License-Identifier : MPL-2.0
+-- Maintainer              : Haskell Beginners 2022 Course <haskell.beginners2022@gmail.com>
+-- Stability               : Stable
+-- Portability             : Portable
+--
+-- Exercises for the Lecture 1 of the Haskell Beginners course.
+--
+-- To complete exercises, you need to complete implementation and add
+-- missing top-level type signatures. You can implement any additional
+-- helper functions. But you can't change the names of the given
+-- functions.
+--
+-- Comments before each function contain explanations and example of
+-- arguments and expected returned values.
+--
+-- It's absolutely okay if you feel that your implementations are not
+-- perfect. You can return to these exercises after future lectures and
+-- improve your solutions if you see any possible improvements.
 module Lecture1
-    ( makeSnippet
-    , sumOfSquares
-    , lastDigit
-    , minmax
-    , subString
-    , strSum
-    , lowerAndGreater
-    ) where
-
-{- | Specify the type signature of the following function. Think about
-its behaviour, possible types for the function arguments and write the
-type signature explicitly.
--}
+  ( makeSnippet,
+    sumOfSquares,
+    lastDigit,
+    minmax,
+    subString,
+    strSum,
+    lowerAndGreater,
+  )
+where
+
+-- | Specify the type signature of the following function. Think about
+-- its behaviour, possible types for the function arguments and write the
+-- type signature explicitly.
+makeSnippet :: Int -> [Char] -> [Char]
 makeSnippet limit text = take limit ("Description: " ++ text) ++ "..."
 
-{- | Implement a function that takes two numbers and finds sum of
-their squares.
-
->>> sumOfSquares 3 4
-25
+-- | Implement a function that takes two numbers and finds sum of
+-- their squares.
+--
+-- >>> sumOfSquares 3 4
+-- 25
+--
+-- >>> sumOfSquares (-2) 7
+-- 53
+--
+-- Explanation: @sumOfSquares 3 4@ should be equal to @9 + 16@ and this
+-- is 25.
 
->>> sumOfSquares (-2) 7
-53
-
-Explanation: @sumOfSquares 3 4@ should be equal to @9 + 16@ and this
-is 25.
--}
 -- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-sumOfSquares x y = error "TODO!"
-
-{- | Implement a function that returns the last digit of a given number.
-
->>> lastDigit 42
-2
->>> lastDigit (-17)
-7
+sumOfSquares :: Num a => a -> a -> a
```

Comment:
`Num` is explained in the following lectures, so you are one step ahead 😸
In here `Int -> Int -> Int` or `Integer -> Integer -> Integer` would also work, as a more specific type 🙂